lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp

```diff
@@ -131,17 +131,54 @@ struct ReduceOpConversion
     }
   }
 
+  void getWriteIndexBasic(ConversionPatternRewriter &rewriter, Location loc,
+                          Attribute layout, SmallVector<Value> &index,
+                          SmallVector<Value> &writeIdx,
+                          std::map<int, Value> &ints, unsigned axis) const {
+    writeIdx = index;
+    auto sizePerThread = triton::gpu::getSizePerThread(layout);
+    Value axisSizePerThread = ints[sizePerThread[axis]];
+    Value _8 = ints[8];
+    Value _16 = ints[16];
+    if (layout.isa<BlockedEncodingAttr>()) {
+      writeIdx[axis] = udiv(index[axis], axisSizePerThread);
+    }
+    auto mmaLayout = layout.dyn_cast<MmaEncodingAttr>();
+    if (mmaLayout && mmaLayout.isAmpere()) {
+      if (axis == 0) {
+        // Because warpTileSize = [16, 8] and threadsPerWarp = [8, 4], each 8
+        // rows in smem would correspond to a warp. The mapping
+        // is: (warp_index) x 8 + (row index within warp)
+        writeIdx[axis] =
+            add(mul(udiv(index[axis], _16), _8), urem(index[axis], _8));
+      } else {
+        // A single thread owns axisSizePerThread contiguous values
+        // on the reduction axis, so after within thread reduction,
+        // writeIdx[axis] = index[axis] / axisSizePerThread
+        writeIdx[axis] = udiv(index[axis], axisSizePerThread);
+      }
+    }
+    if (mmaLayout && !mmaLayout.isAmpere()) {
+      llvm::report_fatal_error("Unsupported layout");
```

Comment:
I noticed we are using `llvm_unreachable` and `llvm::report_fatal_error` interchangeably, or maybe I didn't notice the pattern. From the documentation, it seemed to me that report_fatal_error would be the right one here, what do you think ?

Reply:
Yep I agree. We should use llvm::report_fatal_error